fastlane/Fastfile

```diff
@@ -64,26 +72,55 @@ platform :ios do
+  lane :release_qa do
+    release build_configuration: :qa
+  end
+
+
+  desc "Releases a new version to `TestFlight`. This lane must receives the following parameters:"
+  desc "- build_configuration: A build configuration to deploy. Can be any of: %s" % [:test, :qa, :testflight, :release].to_s
+  desc "- bump_type (optional): represents the type of deploy. If not specified, the user will be asked for it. 
+  Its allowed values depend on the configuration: %s" % build_configuration_allowed_bump_types.to_s
   desc ""
   desc "It has basically 3 main responsabilities: build/version number managing, app building, and deploy."
   desc ""
-  desc "- Reads the build/version number from `'CURRENT_BUILD_NUMBER'/'CURRENT_VERSION_NUMBER'` 
-xcode user defined build configuration for the configuration in use."
-  desc "- Increments and saves these new values in the xcode user defined build configuration."
-  desc "- Sets these incremented build and version values in the `Info.plist` to be used to build the app."
+  desc "- Reads the version number from `CURRENT_VERSION_NUMBER'` 
+  xcode user defined build configuration for the configuration in use."
+  desc "- Gets the latest build number for that version in TestFlight"
+  desc "- Increments and saves the new version in the xcode user defined build configuration."
+  desc "- Sets these incremented version values in the `Info.plist` to be used to build the app."
   desc "- Builds the app using `gym` and `match` to get the signing identity. The provisioning profile in use is the one \
-selected in xcode for the configuration mapped by `:testflight`."
+selected in xcode for the selected configuration"
   desc "- Uploads the generated `.dsym` file to `Rollbar`."
   desc "- Discards the changes in `Info.plist`. Given this file is used for every configuration, these values are stored \
 in xcode user defined build configuration and just reflected in `Info.plist` during building."
   desc "- Uploads the application to `TestFlight` using `pilot`."
   desc ""
-  desc "Receives a parameter `bump_type` representing the type of deploy. It can be any of %s" % release_bump_types.to_s
   desc "Check [here](http://semver.org/) for reference about versioning."
-  desc "Build is initialized in %s, and version in %s, so first deploy must be %s" % [first_build, first_version, release_bump_types.last]
-  lane :release_testflight do |options|
+  desc "Build is initialized in %s, and version in %s, so first deploy must be a %s" % [first_build, first_version, all_bump_types.last]
+  private_lane :release do |options|
+
+    build_configuration = options[:build_configuration]
+    allowed_bump_types = build_configuration_allowed_bump_types[build_configuration]
+
+    if options[:bump_type] && allowed_bump_types.include?(options[:bump_type]) == false
+      UI.user_error! "The bump_type specified for this lane can only be one of %s" % allowed_bump_types.to_s
+    end
+
+    desc "Bundle identifier from xCode project"
```

Comment:
`desc "Read bundle identifier from project configuration."` I know it was just moved code, but I think this looks better.